packages/contracts-bedrock/scripts/DeployImplementations.s.sol

```diff
@@ -217,37 +270,136 @@ contract DeployImplementations is Script {
+        deployDisputeGameFactoryImpl(_dsi, _dso);
+
+        // Deploy the OP Stack Manager with the new implementations set.
+        deployOPStackManager(_dsi, _dso);
 
         _dso.checkOutput();
     }
 
     // -------- Deployment Steps --------
 
-    function deploySystemConfigImpl(DeployImplementationsInput, DeployImplementationsOutput _dso) public {
+    // --- OP Stack Manager ---
+
+    function opsmSystemConfigSetter(
+        DeployImplementationsInput,
+        DeployImplementationsOutput _dso
+    )
+        internal
+        view
+        virtual
+        returns (OPStackManager.ImplementationSetter memory)
+    {
+        return OPStackManager.ImplementationSetter({
+            name: "SystemConfig",
+            info: OPStackManager.Implementation(address(_dso.systemConfigImpl()), SystemConfig.initialize.selector)
+        });
+    }
+
+    function createOPSMContract(
+        DeployImplementationsInput _dsi,
+        DeployImplementationsOutput,
+        OPStackManager.Blueprints memory blueprints
+    )
+        internal
+        virtual
+        returns (OPStackManager opsm_)
+    {
+        SuperchainConfig superchainConfigProxy = _dsi.superchainConfigProxy();
+        ProtocolVersions protocolVersionsProxy = _dsi.protocolVersionsProxy();
+
+        vm.broadcast(msg.sender);
+        opsm_ = new OPStackManager({
+            _superchainConfig: superchainConfigProxy,
+            _protocolVersions: protocolVersionsProxy,
+            _blueprints: blueprints
+        });
+    }
+
+    function deployOPStackManager(DeployImplementationsInput _dsi, DeployImplementationsOutput _dso) public virtual {
+        string memory release = _dsi.release();
+
+        // First we deploy the blueprints for the singletons deployed by OPSM.
+        // forgefmt: disable-start
+        bytes32 salt = bytes32(0);
+        OPStackManager.Blueprints memory blueprints;
+
+        vm.startBroadcast(msg.sender);
+        blueprints.addressManager = deployBytecode(Blueprint.blueprintDeployerBytecode(type(AddressManager).creationCode), salt);
+        blueprints.proxy = deployBytecode(Blueprint.blueprintDeployerBytecode(type(Proxy).creationCode), salt);
+        blueprints.proxyAdmin = deployBytecode(Blueprint.blueprintDeployerBytecode(type(ProxyAdmin).creationCode), salt);
+        blueprints.l1ChugSplashProxy = deployBytecode(Blueprint.blueprintDeployerBytecode(type(L1ChugSplashProxy).creationCode), salt);
+        blueprints.resolvedDelegateProxy = deployBytecode(Blueprint.blueprintDeployerBytecode(type(ResolvedDelegateProxy).creationCode), salt);
+        vm.stopBroadcast();
+        // forgefmt: disable-end
+
+        // This call contains a broadcast to deploy OPSM.
+        OPStackManager opsm = createOPSMContract(_dsi, _dso, blueprints);
+
+        OPStackManager.ImplementationSetter[] memory setters = new OPStackManager.ImplementationSetter[](5);
+        setters[0] = OPStackManager.ImplementationSetter({
```

Comment:
Should this array also contain `DelayedWETH` and `DisputeGameFactory`?

Reply:
It will eventually, we are going to add Fault Proof support in a separate PR. Interop devnet 0 (which this PR unblocks) does not need FPs, so we just stubbed it in this PR

Reply:
Cool, tracking this in: https://github.com/ethereum-optimism/optimism/issues/11815